### PR TITLE
Filter percepts

### LINKED ISSUE #64: Toggling percepts to enable goal "percept mocking"
It is currently possible to "mock" a percept for testing purposes in the goal test2g files. But since the tygron environment keeps sending the actual percepts this isn't usable yet.
We should create a action for the agent so it can toggle certain percepts.

### PR DESCRIPTION
**Pull request by Denpeer**
This PR will allow the agent to use the filter_percepts action to temporally disable certain percepts. resolves #64

> The filter_percepts(PerceptList) action can be called with as only parameter
> a list of percepts that won't be send to the agent in the next cycle.
> The connector will also send a response percept: filtered_percepts([PerceptList])
> with a list of percepts that are currently being filtered by the connector.
> 
> The filter_percepts(PerceptList) action requires a complete list of percepts to be filtered,
> any percepts not included will not be filtered even if they were previously.
> This allows you to re-enable all percepts by sending a empty list as parameter.

The [internal pr](https://github.com/Denpeer/tygron/pull/23) can be found at denpeer/tygron

User Story:
As a developer I want to be able to disable certain percepts so that I can test my goal agent code without interference from the tygron environment.

Travis: [Build Status](https://travis-ci.org/Denpeer/tygron)

### CODE REVIEW

**Comment by jannelie** [comment]
We approve -CTD

**Comment by lisv** [comment]
Seems handy to increase the performance as percepts that will not be used anyways only slow down the agent. So we approve.
-CodeFox

File: contextvh/src/main/java/contextvh/ContextEntity.java [hunk 1]
```diff
@@ -30,6 +36,7 @@
 	 */
 	public ContextEntity(final EntityListener env, final String intendedStakeholder, final Integer slotID) {
 		super(env, intendedStakeholder, slotID);
+		customActions.addAction(perceptsFilter);
```
Comment (spiderbiggen) [inline contextvh/src/main/java/contextvh/ContextEntity.java]: Why did you add this here and not in the action container's constructor

Comment (Denpeer) [inline contextvh/src/main/java/contextvh/ContextEntity.java]: @spiderbiggen  This works with relevantarea's because it's only necessary when the action is called (and thus given as parameter).FilterPercepts is also needed when other percepts are send and the action hasn't been called for a while.

See our [internal pr](https://github.com/Denpeer/tygron/pull/23) for more information

Comment (Denpeer) [inline contextvh/src/main/java/contextvh/ContextEntity.java]: @spiderbiggen on second thought, this is possible since `ActionContainer` extends Hashmap.
See 57e1e15

**Comment by spiderbiggen** [comment]
We Approve

-RAMpestampers

**Comment by levilime** [comment]
I think it's a cool idea, I've only reviewed it statically which I approve, I trust other teams have also tested this dynamically
